app/controllers/openid_application_controller.rb

```diff
@@ -51,7 +51,28 @@ def token_from_request
     auth_request = request.authorization.to_s
     return unless auth_request[TOKEN_REGEX]
 
-    Token.new(auth_request.sub(TOKEN_REGEX, '').gsub(/^"|"$/, ''), fetch_aud)
+    token_string = auth_request.sub(TOKEN_REGEX, '').gsub(/^"|"$/, '')
+
+    if jwt?(token_string)
+      Token.new(token_string, fetch_aud)
+    else
+      # Future block for opaque tokens
+      raise error_klass('Opaque tokens not supported.')
```

Comment:
I recommend we change this error message to one of two things.

- A copy of what gets returned today for invalid JWTs 
- A generic error (e.g., `Invalid token`)

My only concern calling out opaque tokens specifically is that some APIs return this to the client and it's unnecessarily detailed. They need to be aware of the types of tokens that are valid, just that what they have used is invalid.

Reply:
Understood & agreed.  

Invalid JWTs to date return segment encoding errors along with the 401.  Since that wouldn't apply, I went with the `Invalid token` suggestion.